lib/bindConnectors/formatInputBody/date.js

```diff
@@ -0,0 +1,53 @@
+/*
+* Auto format dates according to the format that the schema property wants.
+*/
+var moment = require('moment');
+var isUnixSecondTimestamp = require('../../utils/isUnixSecondTimestamp');
+var isUnixMsTimestamp = require('../../utils/isUnixMsTimestamp');
+var isValidDate = require('../../utils/isValidDate');
+
+
+module.exports = function (value, schemaProperty) {
+
+  var allowedTypes = ['string', 'number'];
+  var allowedDateFormats = ['date', 'time', 'datetime'];
+
+  if (allowedTypes.indexOf(schemaProperty.type) !== -1 &&
+      allowedDateFormats.indexOf(schemaProperty.format) !== -1 &&
+      _.isString(schemaProperty.date_mask) &&
+      (_.isString(value) || _.isNumber(value))
+  ) {
+
+    var dateValue = value;
+
+    // If a unix timestamp in seconds is inputted, x1000 before inputting here
+    if (isUnixSecondTimestamp(dateValue)) {
+      dateValue = Number(dateValue) * 1000;
+    }
+
+    else if (isUnixMsTimestamp(dateValue)) {
+      dateValue = Number(dateValue);
+    }
+
+    // Throw an error if an invalid date has been inputted
+    if (!isValidDate(dateValue)) {
```

Comment:
nope - it throws loads of warnings into the console, which will end up getting captured when running in Lambda. (i.e. our logs bill will go up)